odbc/src/api/functions.rs

```diff
@@ -337,13 +337,82 @@ fn sql_alloc_handle(
 #[no_mangle]
 pub unsafe extern "C" fn SQLBindCol(
     hstmt: HStmt,
-    _col_number: USmallInt,
-    _target_type: SmallInt,
-    _target_value: Pointer,
-    _buffer_length: Len,
-    _length_or_indicatior: *mut Len,
+    col_number: USmallInt,
+    target_type: SmallInt,
+    target_value: Pointer,
+    buffer_length: Len,
+    length_or_indicator: *mut Len,
 ) -> SqlReturn {
-    unimpl!(hstmt);
+    panic_safe_exec_keep_diagnostics!(
+        debug,
+        || {
+            let mongo_handle = MongoHandleRef::from(hstmt);
+            let stmt = must_be_valid!((*mongo_handle).as_statement());
+
+            // Currently, we only support column binding of one row at a time.
+            if stmt.attributes.read().unwrap().row_array_size != 1
+                || !stmt
+                    .attributes
+                    .read()
+                    .unwrap()
+                    .row_bind_offset_ptr
+                    .is_null()
+                || stmt.attributes.read().unwrap().row_bind_type
+                    != (BindType::SQL_BIND_BY_COLUMN as usize)
+            {
+                unsupported_function!(hstmt)
```

Comment:
Is it possible/worth providing the context that leads to this panic? As in, which of the three conditions actually triggered this?

Reply:
Yes, I think that would be helpful! Originally, Natacha told me to just return `unsupported_function!(hstmt)` if one of these conditions was encountered; however, it would be helpful if the user knew what the actually issue is. Also, I just noticed that `unsupported_function!(hstmt)` will run this:
```
let name = function_name!();
add_diag_info!(mongo_handle, ODBCError::Unimplemented(name));
```
So the user will get a message essentially saying `SQLBindCol` is unimplemented. 

I could do something like this instead:
```
add_diag_info!(mongo_handle, ODBCError::Unimplemented("<describe the incorrect setting>"));
return SqlReturn::ERROR;
```
This way the user knows what went wrong.

Reply:
Nice!